MEN-8723 - chore(gui): moved deployment retry setting to setting section

mzedel opened the pull request:
- this is to reduce confusion & clarify the permanent "setting" character

Ticket: MEN-8723

~this needs: https://github.com/NorthernTechHQ/nt-gui/pull/731~

in deployment creation:
<img width="478" height="106" alt="Screenshot 2026-05-22 at 14 27 41" src="https://github.com/user-attachments/assets/ba42971b-9900-4d71-978c-370ec48c361d" />

in settings:
<img width="544" height="169" alt="Screenshot 2026-05-22 at 14 27 52" src="https://github.com/user-attachments/assets/9dba5788-1ec9-481b-b64e-b3234deaaf47" />

mzedel committed:
fix(gui): ensured internal links can be opened in a new tab too

Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

mzedel committed:
chore(gui): moved deployment retry setting to setting section

- this is to reduce confusion & clarify the permanent "setting" character

Ticket: MEN-8723
Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

mineralsfree reviewed (changes requested):
I think we need to take care of the limits, as the current `max` in `NumberInput` only affects the stepper being disabled. Others are minor

mzedel committed:
test(gui): aligned deployment creation tests w/ switch to slide out panel & updated retry handling

Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

mzedel committed:
chore(gui): moved deployment retry setting to setting section

- this is to reduce confusion & clarify the permanent "setting" character

Ticket: MEN-8723
Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>

mzedel committed:
chore(gui): updated snapshots

Signed-off-by: Manuel Zedel <manuel.zedel@northern.tech>
